### DISCUSSION TITLE
Calling Supabase Edge Functions from postgres functions: What is the recommended practice?

### DISCUSSION THREAD

**Discussion by TheLastJoel** [supabase/supabase#28341] [Questions] [answered]
Hey. I have a Supabase edge function which calls my ai-player API. (This is for a turn based game I am making)

At present, the edge function (which generates a new move) is called on every insert to my 'moves' table.
The API does not need to be called if the previous move was by the ai-player itself, so currently my Edge Function fetches the player who made the move based off the move.playerid and checks whether it is AI or not.
There are also a few other similar validations that happen before to determine if the API is called or not.

**I wondered if a better way is to have a postgres function which does the initial validations and then calls the edge function only if the validation passes.**

1) Would this be more / less efficient?
2) Is this good practice?
3) Is it actually worth doing this?

Thanks!

**Comment by GaryAustin1** [comment, marked as answer]
Not commenting on what you want to check in SQL as you can check almost anything in the db in a trigger function.  You can use pg_net or http extension to call your edge function.  Webhooks, the Supabase method uses pg_net.
